### PR TITLE
Roster prototype

### PR DESCRIPTION
**Pull request by mifrdlik**
#### Description of changes

Introducing a calling roster prototype. The purpose of this prototype is to monitor potential performance regression (and confirm the effect of performance optimization) across Fluent versions.

roster-model mov

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/12180)

### CODE REVIEW

**Comment by pompomon** [comment]
@mifrdlik , thank you for the contribution!
Please could you place this example under packages/fluentui/docs/src/examples/components/Tree/Performance/RosterTree.perf.tsx and use single file (please see packages/fluentui/docs/src/examples/components/Box/Performance/BoxMinimal.perf.tsx or packages/fluentui/docs/src/examples/components/Tree/Performance/TreeWith60ListItems.perf.tsx as the example), so we can automatically measure performance

Cc. @layershifter @miroslavstastny

File: packages/fluentui/docs/src/prototypes/Roster/Roster.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  const presenters = getSection('presenters', Array.from(_presenters.values()).sort(lexCompareData));
+  const attendees = getSection('attendees', Array.from(_attendees.values()).sort(lexCompareData));
+  const suggestions = getSection('suggestions', Array.from(_suggestions.values()).sort(lexCompareData));
+
+  const actions = useRosterActions(rosterData, setRosterData);
+
+  return (
+    <ActionsContext.Provider value={actions}>
+      <Tree
+        as="div"
```
Comment (jurokapsiar) [inline packages/fluentui/docs/src/prototypes/Roster/Roster.tsx]: Tree renders a div by default

File: packages/fluentui/docs/src/prototypes/Roster/RosterState.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
+import { Icon, Flex } from '@fluentui/react';
+
+export const RosterState: React.FunctionComponent<{ action: React.ReactNode; isMuted: boolean }> = ({ action, isMuted }) => {
+  return (
+    <Flex vAlign="center">
+      <Flex>{isMuted ? <Icon outline name={'mic-off'} xSpacing="both" /> : null}</Flex>
```
Comment (assuncaocharles) [inline packages/fluentui/docs/src/prototypes/Roster/RosterState.tsx]:
```suggestion
      <Flex>{isMuted ? <Icon outline name="mic-off" xSpacing="both" /> : null}</Flex>
```

File: packages/fluentui/docs/src/prototypes/Roster/RosterTitle.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import * as React from 'react';
+import { RosterSectionType } from './interface/roster.interface';
+import { Flex, Icon } from '@fluentui/react';
+import { rosterTitleIconStyles } from './styles/styles';
+
+export const RosterSectionTitle: React.FunctionComponent<{ type: RosterSectionType }> = ({ type }) => {
+  const [isToggled, setToggled] = React.useState(true);
+  return (
+    <Flex onClick={() => setToggled(!isToggled)}>
```
Comment (assuncaocharles) [inline packages/fluentui/docs/src/prototypes/Roster/RosterTitle.tsx]:
```suggestion
    <Flex onClick={() => setToggled(toggled => !toggled)}>
```

File: packages/fluentui/docs/src/prototypes/Roster/RosterTitle.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+      <Icon name={isToggled ? 'icon-arrow-down' : 'icon-arrow-end'} styles={rosterTitleIconStyles} />
+      {(type as string).charAt(0).toUpperCase() + (type as string).slice(1)}
```
Comment (assuncaocharles) [inline packages/fluentui/docs/src/prototypes/Roster/RosterTitle.tsx]:
```suggestion
      {type.charAt(0).toUpperCase() + type.slice(1)}
```

File: packages/fluentui/docs/src/prototypes/Roster/RosterTitle.tsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import * as React from 'react';
+import { RosterSectionType } from './interface/roster.interface';
+import { Flex, Icon } from '@fluentui/react';
```
Comment (assuncaocharles) [inline packages/fluentui/docs/src/prototypes/Roster/RosterTitle.tsx]:
```suggestion
import { Flex, Icon } from '@fluentui/react-northstar';
```

**Comment by mifrdlik** [comment]
Closing as there's the exact same thing already implemented in examples/components/Tree/Performance that I didn't notice.

**Pull request closed without merging**